packages/nocodb/src/db/BaseModelSqlv2.ts

```diff
@@ -2193,7 +2193,19 @@ class BaseModelSqlv2 {
         await this.beforeInsert(insertObj, trx, cookie);
       }
 
-      await this.model.getColumns();
+      const columns = await this.model.getColumns();
+
+      // exclude auto increment columns in body
+      for (const col of columns) {
+        if (col.ai === true) {
+          const keyName =
+            data?.[col.column_name] !== undefined ? col.column_name : col.title;
+
+          if (data[keyName]) {
+            delete data[keyName];
```

Comment:
You are just delete=ing it from `data` and not from `insertObj` so still it will insert value.

Reply:
fixed and tested